cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -405,155 +406,149 @@ runQueryKesPeriodInfo (AnyConsensusModeParams cModeParams) network nodeOpCertFil
-     then
-       return ( slotsTillNewKesKey
-              , currentKesPeriod
-              , kesPeriodIntervalStart
-              , kesPeriodIntervalEnd
-              , SuccessDiagnostic OpCertCurrentKesPeriodWithinInterval
-              )
-     else do
-       let fd = FailureDiagnostic $ OpCertKesPeriodOutsideOfCurrentInterval
-                                      (nodeOpCertFile, opCertKesPeriod)
-                                      kesPeriodIntervalStart
-                                      kesPeriodIntervalEnd
-       return ( 0
-              , currentKesPeriod
-              , kesPeriodIntervalStart
-              , kesPeriodIntervalEnd
-              , fd
-              )
+     -> OpCertEndingKesPeriod
+     -> Maybe UTCTime
+   opCertExpiryUtcTime eInfo gParams (OpCertEndingKesPeriod oCertExpiryKesPeriod) =
+     let time = epochInfoSlotToUTCTime
+                  eInfo
+                  (SystemStart $ protocolParamSystemStart gParams)
+                  (fromIntegral $ oCertExpiryKesPeriod * fromIntegral (protocolParamSlotsPerKESPeriod gParams))
+     in case time of
+          Left _ -> Nothing
+          Right t -> Just t
+
+   renderOpCertNodeAndOnDiskCounterInformation :: FilePath -> OpCertNodeAndOnDiskCounterInformation -> IO ()
+   renderOpCertNodeAndOnDiskCounterInformation opCertFile opCertCounterInfo =
+     case opCertCounterInfo of
+      OpCertOnDiskCounterMoreThanOrEqualToNodeState _ _ ->
+        putStrLn @String "✓ The operational certificate counter agrees with the node protocol state counter"
+      OpCertOnDiskCounterBehindNodeState onDiskC nodeStateC ->
+        putStrLn $ "✗ The protocol state counter is greater than the counter in the operational certificate at: " <> opCertFile <> "\n" <>
+                   "On disk operational certificate counter: " <> show (unOpCertOnDiskCounter onDiskC) <> "\n" <>
+                   "Protocol state counter: " <> show (unOpCertNodeStateCounter nodeStateC)
+      OpCertNoBlocksMintedYet (OpCertOnDiskCounter onDiskC) ->
+        putStrLn $ "✗ No blocks minted so far with the operational certificate at: " <> opCertFile <> "\n" <>
+                   "On disk operational certificate counter: " <> show onDiskC
+
+
+   createQueryKesPeriodInfoOutput
+     :: OpCertIntervalInformation
+     -> OpCertNodeAndOnDiskCounterInformation
+     -> EpochInfo (Either Text)
+     -> GenesisParameters
+     -> O.QueryKesPeriodInfoOutput
+   createQueryKesPeriodInfoOutput oCertIntervalInfo oCertCounterInfo eInfo gParams  =
+     let (_s, e, _c, mStillExp) = case oCertIntervalInfo of
+                                  OpCertWithinInterval st end curr sTillExp -> (st, end, curr, Just sTillExp)
+                                  OpCertStartingKesPeriodIsInTheFuture st end curr -> (st, end, curr, Nothing)
+                                  OpCertExpired st end curr -> (st, end, curr, Nothing)
+                                  OpCertSomeOtherError st end curr -> (st, end, curr, Nothing)
```

Comment:
Why does the case return `st` and `curr` only to ignore them?

Reply:
This was prior to a refactor I did. I will adjust the case to return a tuple.